Tidy up index.rst, from sylabs 38

edytuk opened the pull request:
This pulls in sylabs PR

- sylabs/singularity-userdocs#38

The original PR description was:

> The initial index page was an overwhelming TOC, spanning many
> pages. Cut down the levels of toctree shown, and add some explanatory
> text for the sections.

DrDaveD commented on index.rst:
```diff
@@ -1,83 +1,120 @@
 Sharing & Online Services
 =========================
 
+HPCng offers a suite of container services, with a free tier and
```
No, it is Sylabs that offers those services.  Just leave this part of the changes out.

edytuk replied:
I removed paragraph

DrDaveD commented on index.rst:
```diff
@@ -1,83 +1,116 @@
 
    Remote Endpoints <endpoint>
-   Cloud Library <cloud_library>
+   HPCng Cloud Library <cloud_library>
```
This would should not say HPCng.

edytuk replied:
I updated it to Sylabs

DrDaveD commented:
When you submit this for real, make it just one commit.

edytuk committed:
Tidy up index.rst

The initial index page was an overwhelming TOC, spanning many
pages. Cut down the levels of toctree shown, and add some explanatory
text for the sections.

edytuk commented:
I updated this PR in order to have one commit
